crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -4065,4 +4148,101 @@ pub(crate) mod tests {
         // The waiting should successfully complete.
         wait.await.unwrap();
     }
+
+    #[async_test]
+    async fn get_room_settings_returns_none_for_unknown_room() {
+        let machine = OlmMachine::new(user_id(), alice_device_id()).await;
+
+        machine
+            .set_room_settings(room_id!("!test:localhost"), &RoomSettings::default())
+            .await
+            .unwrap();
```

Comment:
This code is not necessary for this test. I understand the idea, but it looks we can safely remove it.